crypto/src/main/java/net/consensys/cava/crypto/sodium/Signature.java

```diff
@@ -502,4 +526,63 @@ public static boolean verifyDetached(byte[] message, byte[] signature, Signature
 
     return true;
   }
+
+  /**
+   * Signs a message for a given key.
+   *
+   * @param message The message to sign.
+   * @param secretKey The secret key to sign the message with.
+   * @return The signature prepended to the message
+   */
+  public static Bytes sign(Bytes message, Signature.SecretKey secretKey) {
+    return Bytes.wrap(sign(message.toArrayUnsafe(), secretKey));
+  }
+
+  /**
+   * Signs a message for a given key.
+   *
+   * @param message The message to sign.
+   * @param secretKey The secret key to sign the message with.
+   * @return The signature prepended to the message
+   */
+  public static byte[] sign(byte[] message, Signature.SecretKey secretKey) {
+    byte[] signature = new byte[(int) Sodium.crypto_sign_bytes() + message.length];
+    LongLongByReference signatureLengthReference = new LongLongByReference();
+    int rc = Sodium.crypto_sign(signature, signatureLengthReference, message, message.length, secretKey.bytesArray());
+    if (rc != 0) {
+      throw new SodiumException("crypto_sign: failed with result " + rc);
+    }
+
+    return signature;
+  }
+
+  /**
+   * Verifies the signature of the signed message using the public key and returns the message.
+   * 
+   * @param signed signed message (signature + message)
+   * @param publicKey pk used to verify the signature
+   * @return the message
+   */
+  public static Bytes openSigned(Bytes signed, Signature.PublicKey publicKey) {
```

Comment:
`openSigned` seems like the wrong name here. The existing methods are `signDetached` and `verifyDetached`. Should these be simply `sign` and `verify`?

Reply:
Yes, `verify` is better and more readable.  Changed.